packages/wdio-junit-reporter/src/index.ts

```diff
@@ -304,6 +304,13 @@ class JunitReporter extends WDIOReporter {
     private _format (val: any) {
         return JSON.stringify(limit(val))
     }
+
+    private _sameFileName(file1: string, file2: string) {
+        // ensure both files are not a file URL
+        file1 = file1 && file1.startsWith('file://') ? url.fileURLToPath(file1) : file1
+        file2 = file2 && file2.startsWith('file://') ? url.fileURLToPath(file2) : file2
```

Comment:
Your type definition for the function suggest that `file1` and `file2` are always a string so none need to check for their existence
```suggestion
        file1 = file1.startsWith('file://') ? url.fileURLToPath(file1) : file1
        file2 = file2.startsWith('file://') ? url.fileURLToPath(file2) : file2
```

Reply:
that was my initial code, but the tests failed because in the tests it may be null, so I add the condition to be sure as I conclued that it may introduce a regression where sometimes the filename can be null.

Reply:
We can use the example?.myProperty  notation here instead

Reply:
I think we should update the test then to confirm with the typing.

Reply:
I added the ? in both parameter declaration and usage, and a unit test on the function _sameFileName